app/package/controllers.py

```diff
@@ -330,70 +331,66 @@ def purge_data_package(publisher, package):
         return handle_error('GENERIC_ERROR', e.message, 500)
 
 
-@package_blueprint.route("/<publisher>/<package>/finalize",
-                         methods=["POST"])
+@package_blueprint.route("/upload", methods=["POST"])
 @requires_auth
-@is_allowed('Package::Create')
```

Comment:
Ah ok, i now see that below you actually have moved authorization in method (something i agree with and recommended you would want to do back in December!)